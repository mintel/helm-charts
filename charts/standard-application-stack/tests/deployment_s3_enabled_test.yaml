suite: Test S3 Enabled Deployment
templates:
  - deployment.yaml
release:
  namespace: test-namespace
tests:
  - it: Check S3 envfrom secretRef is present
    set:
      global.clusterEnv: qa
      global.name: test-app
      s3.enabled: true
      # Disable default application secret
      externalSecret.enabled: false
    asserts:
      - isKind:
          of: Deployment
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: spec.template.spec.containers[0].envFrom[0].secretRef.name
          value: test-app-s3
  - it: Check S3 envfrom secretRef is present for multiple instances of TF Cloud helm chart
    set:
      global.clusterEnv: qa
      global.name: test-app
      global.terraform.externalSecrets: true
      global.clusterName: cluster1
      global.clusterRegion: eu-west-1
      global.owner: sre
      global.partOf: test-project
      s3:
        enabled: true
        outputSecret: true
        terraform:
          instances:
            test-bucket: {}
            test-bucket-another: {}
      # Disable default application secret
      externalSecret.enabled: false
    asserts:
      - isKind:
          of: Deployment
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: spec.template.spec.containers[0].envFrom[0].secretRef.name
          value: mntl-test-bucket-s3
      - equal:
          path: spec.template.spec.containers[0].envFrom[1].secretRef.name
          value: mntl-test-bucket-another-s3
  - it: Check S3 secretName Override
    set:
      global.clusterEnv: qa
      global.name: test-app
      s3.enabled: true
      s3.secretNameOverride: overrideName
      # Disable default application secret
      externalSecret.enabled: false
    asserts:
      - isKind:
          of: Deployment
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: spec.template.spec.containers[0].envFrom[0].secretRef.name
          value: overrideName
      - equal:
          path: metadata.annotations.[secret.reloader.stakater.com/reload]
          value: overrideName
  - it: Check S3 envfrom secretRef is present for local environment
    set:
      global.clusterEnv: local
      global.name: test-app
      s3.enabled: true
      # Disable default application secret
      externalSecret.enabled: false
    asserts:
      - isKind:
          of: Deployment
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.annotations.[secret.reloader.stakater.com/reload]
          value: test-app-s3
      - equal:
          path: spec.template.spec.containers[0].envFrom[0].secretRef.name
          value: test-app-s3
