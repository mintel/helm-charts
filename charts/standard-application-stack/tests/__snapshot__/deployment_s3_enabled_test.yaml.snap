Check S3 envfrom secretRef is present:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        secret.reloader.stakater.com/reload: test-app-s3
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/application: test-app
        app.mintel.com/component: test-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      minReadySeconds: 10
      replicas: 2
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: test-app
      strategy:
        rollingUpdate:
          maxSurge: 15%
          maxUnavailable: 10%
        type: RollingUpdate
      template:
        metadata:
          labels:
            app.kubernetes.io/component: app
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: test-app
            app.mintel.com/application: test-app
            app.mintel.com/component: test-app
            app.mintel.com/env: qa
            app.mintel.com/region: ${CLUSTER_REGION}
            name: test-app
        spec:
          containers:
            - command:
                - /app/docker-entrypoint.sh
              env:
                - name: KUBERNETES_POD_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
                - name: APP_ENVIRONMENT
                  value: qa
                - name: RUNTIME_ENVIRONMENT
                  value: kubernetes
              envFrom:
                - secretRef:
                    name: test-app-s3
              image: registry.gitlab.com/test:v0.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                periodSeconds: 10
                timeoutSeconds: 1
              name: main
              ports:
                - containerPort: 8000
                  name: http
              readinessProbe:
                httpGet:
                  path: /readiness
                  port: http
                  scheme: HTTP
                periodSeconds: 10
                timeoutSeconds: 1
              resources:
                limits: {}
                requests: {}
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                runAsNonRoot: true
                runAsUser: 1000
              startupProbe:
                failureThreshold: 60
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                periodSeconds: 5
          imagePullSecrets:
            - name: image-pull-gitlab
            - name: image-pull-docker-hub
          securityContext:
            runAsNonRoot: true
            runAsUser: 1000
          serviceAccountName: test-app
          terminationGracePeriodSeconds: 30
          topologySpreadConstraints:
            - labelSelector:
                matchLabels:
                  app.kubernetes.io/component: app
                  app.kubernetes.io/name: test-app
              matchLabelKeys:
                - pod-template-hash
              maxSkew: 1
              topologyKey: topology.kubernetes.io/zone
              whenUnsatisfiable: DoNotSchedule
Check S3 envfrom secretRef is present for local environment:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        secret.reloader.stakater.com/reload: test-app-s3
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/application: test-app
        app.mintel.com/component: test-app
        app.mintel.com/env: local
        app.mintel.com/region: local
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      minReadySeconds: 10
      replicas: 2
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: test-app
      strategy:
        rollingUpdate:
          maxSurge: 15%
          maxUnavailable: 10%
        type: RollingUpdate
      template:
        metadata:
          labels:
            app.kubernetes.io/component: app
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: test-app
            app.mintel.com/application: test-app
            app.mintel.com/component: test-app
            app.mintel.com/env: local
            app.mintel.com/region: local
            name: test-app
        spec:
          containers:
            - command:
                - /app/docker-entrypoint.sh
              env:
                - name: KUBERNETES_POD_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
                - name: INIT_DB
                  value: "1"
                - name: SENTRY_DONT_SEND_EVENTS
                  value: "1"
                - name: DEBUG
                  value: "1"
                - name: APP_ENVIRONMENT
                  value: local
                - name: RUNTIME_ENVIRONMENT
                  value: kubernetes
              envFrom:
                - secretRef:
                    name: test-app-s3
              image: k3d-default.localhost:5000/test:v0.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                periodSeconds: 10
                timeoutSeconds: 1
              name: main
              ports:
                - containerPort: 8000
                  name: http
              readinessProbe:
                httpGet:
                  path: /readiness
                  port: http
                  scheme: HTTP
                periodSeconds: 10
                timeoutSeconds: 1
              resources:
                limits: {}
                requests: {}
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                runAsNonRoot: true
                runAsUser: 1000
              startupProbe:
                failureThreshold: 60
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                periodSeconds: 5
          imagePullSecrets:
            - name: image-pull-gitlab
            - name: image-pull-docker-hub
          securityContext:
            runAsNonRoot: true
            runAsUser: 1000
          serviceAccountName: test-app
          terminationGracePeriodSeconds: 30
Check S3 envfrom secretRef is present for multiple instances of TF Cloud helm chart:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        secret.reloader.stakater.com/reload: mntl-test-bucket-another-s3,mntl-test-bucket-s3
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.kubernetes.io/part-of: test-project
        app.mintel.com/application: test-app
        app.mintel.com/component: test-app
        app.mintel.com/env: qa
        app.mintel.com/owner: sre
        app.mintel.com/region: eu-west-1
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      minReadySeconds: 10
      replicas: 2
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: test-app
          app.kubernetes.io/part-of: test-project
      strategy:
        rollingUpdate:
          maxSurge: 15%
          maxUnavailable: 10%
        type: RollingUpdate
      template:
        metadata:
          labels:
            app.kubernetes.io/component: app
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: test-app
            app.kubernetes.io/part-of: test-project
            app.mintel.com/application: test-app
            app.mintel.com/component: test-app
            app.mintel.com/env: qa
            app.mintel.com/owner: sre
            app.mintel.com/region: eu-west-1
            name: test-app
        spec:
          containers:
            - command:
                - /app/docker-entrypoint.sh
              env:
                - name: KUBERNETES_POD_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
                - name: APP_ENVIRONMENT
                  value: qa
                - name: RUNTIME_ENVIRONMENT
                  value: kubernetes
              envFrom:
                - secretRef:
                    name: mntl-test-bucket-another-s3
                - secretRef:
                    name: mntl-test-bucket-s3
              image: registry.gitlab.com/test:v0.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                periodSeconds: 10
                timeoutSeconds: 1
              name: main
              ports:
                - containerPort: 8000
                  name: http
              readinessProbe:
                httpGet:
                  path: /readiness
                  port: http
                  scheme: HTTP
                periodSeconds: 10
                timeoutSeconds: 1
              resources:
                limits: {}
                requests: {}
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                runAsNonRoot: true
                runAsUser: 1000
              startupProbe:
                failureThreshold: 60
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                periodSeconds: 5
          imagePullSecrets:
            - name: image-pull-gitlab
            - name: image-pull-docker-hub
          securityContext:
            runAsNonRoot: true
            runAsUser: 1000
          serviceAccountName: test-app
          terminationGracePeriodSeconds: 30
          topologySpreadConstraints:
            - labelSelector:
                matchLabels:
                  app.kubernetes.io/component: app
                  app.kubernetes.io/name: test-app
                  app.kubernetes.io/part-of: test-project
              matchLabelKeys:
                - pod-template-hash
              maxSkew: 1
              topologyKey: topology.kubernetes.io/zone
              whenUnsatisfiable: DoNotSchedule
Check S3 secretName Override:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        secret.reloader.stakater.com/reload: overrideName
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/application: test-app
        app.mintel.com/component: test-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      minReadySeconds: 10
      replicas: 2
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: test-app
      strategy:
        rollingUpdate:
          maxSurge: 15%
          maxUnavailable: 10%
        type: RollingUpdate
      template:
        metadata:
          labels:
            app.kubernetes.io/component: app
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: test-app
            app.mintel.com/application: test-app
            app.mintel.com/component: test-app
            app.mintel.com/env: qa
            app.mintel.com/region: ${CLUSTER_REGION}
            name: test-app
        spec:
          containers:
            - command:
                - /app/docker-entrypoint.sh
              env:
                - name: KUBERNETES_POD_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
                - name: APP_ENVIRONMENT
                  value: qa
                - name: RUNTIME_ENVIRONMENT
                  value: kubernetes
              envFrom:
                - secretRef:
                    name: overrideName
              image: registry.gitlab.com/test:v0.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                periodSeconds: 10
                timeoutSeconds: 1
              name: main
              ports:
                - containerPort: 8000
                  name: http
              readinessProbe:
                httpGet:
                  path: /readiness
                  port: http
                  scheme: HTTP
                periodSeconds: 10
                timeoutSeconds: 1
              resources:
                limits: {}
                requests: {}
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                runAsNonRoot: true
                runAsUser: 1000
              startupProbe:
                failureThreshold: 60
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                periodSeconds: 5
          imagePullSecrets:
            - name: image-pull-gitlab
            - name: image-pull-docker-hub
          securityContext:
            runAsNonRoot: true
            runAsUser: 1000
          serviceAccountName: test-app
          terminationGracePeriodSeconds: 30
          topologySpreadConstraints:
            - labelSelector:
                matchLabels:
                  app.kubernetes.io/component: app
                  app.kubernetes.io/name: test-app
              matchLabelKeys:
                - pod-template-hash
              maxSkew: 1
              topologyKey: topology.kubernetes.io/zone
              whenUnsatisfiable: DoNotSchedule
Check S3 stakater secret reloader annotation contains correct secrets for multiple instances of TF Cloud helm chart:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        secret.reloader.stakater.com/reload: mntl-test-bucket-another-s3,mntl-test-bucket-s3,test-app-app,test-app-extra-secret-1
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.kubernetes.io/part-of: test-project
        app.mintel.com/application: test-app
        app.mintel.com/component: test-app
        app.mintel.com/env: qa
        app.mintel.com/owner: sre
        app.mintel.com/region: eu-west-1
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      minReadySeconds: 10
      replicas: 2
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: test-app
          app.kubernetes.io/part-of: test-project
      strategy:
        rollingUpdate:
          maxSurge: 15%
          maxUnavailable: 10%
        type: RollingUpdate
      template:
        metadata:
          labels:
            app.kubernetes.io/component: app
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: test-app
            app.kubernetes.io/part-of: test-project
            app.mintel.com/application: test-app
            app.mintel.com/component: test-app
            app.mintel.com/env: qa
            app.mintel.com/owner: sre
            app.mintel.com/region: eu-west-1
            name: test-app
        spec:
          containers:
            - command:
                - /app/docker-entrypoint.sh
              env:
                - name: KUBERNETES_POD_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
                - name: APP_ENVIRONMENT
                  value: qa
                - name: RUNTIME_ENVIRONMENT
                  value: kubernetes
              envFrom:
                - secretRef:
                    name: test-app-app
                - secretRef:
                    name: mntl-test-bucket-another-s3
                - secretRef:
                    name: mntl-test-bucket-s3
                - secretRef:
                    name: test-app-extra-secret-1
              image: registry.gitlab.com/test:v0.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                periodSeconds: 10
                timeoutSeconds: 1
              name: main
              ports:
                - containerPort: 8000
                  name: http
              readinessProbe:
                httpGet:
                  path: /readiness
                  port: http
                  scheme: HTTP
                periodSeconds: 10
                timeoutSeconds: 1
              resources:
                limits: {}
                requests: {}
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                runAsNonRoot: true
                runAsUser: 1000
              startupProbe:
                failureThreshold: 60
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                periodSeconds: 5
          imagePullSecrets:
            - name: image-pull-gitlab
            - name: image-pull-docker-hub
          securityContext:
            runAsNonRoot: true
            runAsUser: 1000
          serviceAccountName: test-app
          terminationGracePeriodSeconds: 30
          topologySpreadConstraints:
            - labelSelector:
                matchLabels:
                  app.kubernetes.io/component: app
                  app.kubernetes.io/name: test-app
                  app.kubernetes.io/part-of: test-project
              matchLabelKeys:
                - pod-template-hash
              maxSkew: 1
              topologyKey: topology.kubernetes.io/zone
              whenUnsatisfiable: DoNotSchedule
